### ISSUE TITLE
Great Work, but just noticed one thing

### ISSUE THREAD

**Issue by Andrw0830** [hackintosh-stuff/ComboJack#1]
Hello, wanted to thank you for posting this as was searching Github and came across this section. I tried this on my Inspiron 7567 which has the Realtek ALC256 and this is the first solution for the line in mic to show up for headsets! Just to point out as I realized this is only a couple days old and not trying to complain or deter you from improving this, but on my laptop, sound suddenly goes in and out from the headphones with no warning. It's a weird way to describe it as sometimes, sound will also stutter before no sound comes through.  I am using your AppleALC that I found for your XPS just so you know. 

I am just speculating, but when I used to have my HP Envy, I would get similar audio issues, when the line in Mic was being activated, so wonder if that is what's happening in this case. Hope this helps and let me know if you need me to test anything or produce anything to help?

**Comment by maz-1** [comment]
SubsystemID for your sound card ?
Also, do you have working audio under linux?

**Comment by Andrw0830** [comment]
SubsystemID: INTELAUDIO\FUNC_01&VEN_10EC&DEV_0256&SUBSYS_10280798

Yes I do have working audio in Linux with both Speakers/Headphones working.

**Comment by maz-1** [comment]
Did you have working headphone with CodecCommander and proper ssdt?

**Comment by Andrw0830** [comment]
Yes using Codec Commander with a SSDT caused audio to work from headphones to work before just didn't have line in as an input device. I made sure to remove both files when I installed your Combojack. I have attached my audio dump here.
[linux_dump.txt](https://github.com/hackintosh-stuff/ComboJack/files/2619736/linux_dump.txt)

**Comment by maz-1** [comment]
> Yes using Codec Commander with a SSDT caused audio to work from headphones to work before just didn't have line in as an input device. I made sure to remove both files when I installed your Combojack. I have attached my audio dump here.
> [linux_dump.txt](https://github.com/hackintosh-stuff/ComboJack/files/2619736/linux_dump.txt)

OK, your laptop has a subwoofer, I think there is a realtek driver fix-up for that in linux, will dig into it...
Show me the ssdt file for CodecCommander as well?

**Comment by Andrw0830** [comment]
Thanks for looking. I have tried 2 different SSDT's for Codec Commander which allowed headphones to work. The _ALC256 one was provided from someone who owns the same laptop I have while the other one was provided by Insanelydeepak who provided working config for the ALC256. Both SSDT I noticed when used with CodecCommander when resumed from sleep, the codec will make a pop sound after playing media through the headset. This I believes is when the codec is going to sleep.

I am using layout-id=13 for your files and also used the same layout-id when I was using CodecCommander with AppleALC.


[SSDT_ALC256.txt](https://github.com/hackintosh-stuff/ComboJack/files/2622126/SSDT_ALC256.txt)
[SSDT-ALC256.txt](https://github.com/hackintosh-stuff/ComboJack/files/2622127/SSDT-ALC256.txt)

**Comment by Andrw0830** [comment]
Hey man, just was wondering if my information has helped? You mentioned since I have a subwoofer, does this mean I will need my own Layout/Platforms xml and different pinconfig files? If that is the case, I have never gotten or produced those files, so can you help?

Also the pop sound that is heard when the laptop sound is initialized and when resuming from sleep, is there a way to stop that? I don't mind the first pop when the laptop is booting as I can just boot without headphones plugged in, but when the laptop resumes from sleep, the pop happens when I am trying to play media and seems to be sporadic. For example, before going to sleep, the pop usually happens just once (like the audio driver never goes to sleep), but after sleep, the audio driver seems to go to sleep after a few seconds of no media playing which next time produces the pop sound. 

Please let me know what you think and thanks for your work!

**Comment by maz-1** [comment]
Remoe these five lines and build :https://github.com/hackintosh-stuff/ComboJack/blob/a32270fb75d9d5f88d987bc94d763312be23aa24/ComboJack_Source/Combo%20Jack/hda-verb.c#L627
See if the new executable solves your problem

**Comment by Andrw0830** [comment]
Hey I tried building, but get this error in Xcode 10.1

"cp: directory ../../../ComboJack_Installer does not exist
Command PhaseScriptExecution failed with a nonzero exit code"

I looked on google and people think updating cocoapods fixes it, but I tried and still same thing. I copied this repository directly and opened up the Xcode project file in your ComboJack_Source.

**Comment by maz-1** [comment]
I just added device subsystemid detection so correct fixup can be applied to different ALC256 variant. Test if the new executable works well on your system.

**Comment by Andrw0830** [comment]
OK will try, I was just typing when I saw your new message. I did want to update you as before your change just now, when using Apple EarPods, the sound will disappear after 10 seconds when you go into into Sound > Input. Not sure if that happens even on a real Mac as maybe the system doesn't want both Output and Input devices running at the same time.

I have another headset (the one that made the pop sound before) and if you play media and stop playing, within 5 seconds, the audio disengages and no sound can be heard. 

So that was with just removing those lines so will try the new device id change you made and will let you know.

**Comment by Andrw0830** [comment]
OK so I gave this a try and so far so good. Apple EarPods doesn't seem to loose audio when the microphone is activated at least wasn't during my quick tests. I think my other headset that has the pop sound issue must just be a coincidence since Apple EarPods are fine. I mention that because never had that pop happen during when using that headset within Ubuntu. 

Also noticed in your hda-verb.m the //alc256 section still has the 5 lines you mentioned before that I should remove. Is that still necessary with your most recent changes?

**Comment by maz-1** [comment]
> 
> 
> OK so I gave this a try and so far so good. Apple EarPods doesn't seem to loose audio when the microphone is activated at least wasn't during my quick tests. I think my other headset that has the pop sound issue must just be a coincidence since Apple EarPods are fine. I mention that because never had that pop happen during when using that headset within Ubuntu.
> 
> Because of the device-id detection, does that mean I no longer need to remove the 5 lines you advise me to remove before? Thanks again for your assistance and great work!

Yep, the five-line fix will only work on xps 13.

**Comment by Andrw0830** [comment]
> > OK so I gave this a try and so far so good. Apple EarPods doesn't seem to loose audio when the microphone is activated at least wasn't during my quick tests. I think my other headset that has the pop sound issue must just be a coincidence since Apple EarPods are fine. I mention that because never had that pop happen during when using that headset within Ubuntu.
> > Because of the device-id detection, does that mean I no longer need to remove the 5 lines you advise me to remove before? Thanks again for your assistance and great work!
> 
> Yep, the five-line fix will only work on xps 13.

Cool thanks again! Let me know if there is anything else you need me to test.

**Comment by Andrw0830** [comment]
Wanted to ask one thing, are there different types of wired headset mics? I ask because I noticed that while Apple EarPods Linein Mic works, the mic in my other headset doesn't work as just shows line in with the volume adjust slider. Just trying to think of anything that could be causing the sudden pop sound heard while putting the computer to sleep or sometimes when inserting the headset. Sorry as don't want to beat around the bush, but wanting to know if you have different headsets that the mic isn't working or is having weird issues? This headset as mentioned before is BT capable and the Mic works fine when paired via BT or if I hook it up with 3.5mm cable in Ubuntu.

Hope you don't mind me providing these issues to you as just trying to cover everything I noticed. You are the only one so far that has been able to get the headset Mic to work at all so really appreciate it.
